src/test_helpers/gold_stream_evaluator.cc

```diff
@@ -0,0 +1,108 @@
+#include "gold_stream_evaluator.h"
+#include "dtl/dtl.hpp"
+
+#include <sstream>
+#include <vector>
+
+namespace btest {
+
+GoldStreamEvaluator::GoldStreamEvaluator(
+    std::unique_ptr<std::istream> gold_stream,
+    std::unique_ptr<std::istream> actual_stream) {
+  // Take ownership of the streams and verify they are good
+  gold_stream_ = std::move(gold_stream);
+  actual_stream_ = std::move(actual_stream);
+  gold_good_ = gold_stream_->good();
+  actual_good_ = actual_stream_->good();
+}
+
+bool GoldStreamEvaluator::Compare() const {
+  if (!gold_good_ || !actual_good_)
+    throw std::runtime_error("Cannot compare bad streams");
+
+  ResetStreams();
+
+  std::string gold_line;
+  std::string actual_line;
+  bool same = true;
+  unsigned int c1=0, c2=0;
+  
+  // Count the lines of each stream
+  while(!gold_stream_->eof())
+  {
```

Comment:
As we agree on using Google style, please be consistent that curly braces should always be on the last line, not a new line. Also, change all other cases accordingly.